### PR TITLE
Simplify load connection in LocalFilesystemBackend

### PR DESCRIPTION
**Pull request by mik-laj**
Commented inline

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/secrets/local_filesystem.py [hunk 1]
```diff
@@ -235,33 +235,33 @@ def load_variables(file_path: str) -> Dict[str, str]:
...
-def load_connections(file_path: str):
+def load_connections(file_path: str) -> Dict[str, Any]:
     """
     Load connection from text file.
 
     Both ``JSON`` and ``.env`` files are supported.
 
     :return: A dictionary where the key contains a connection ID and the value contains a list of connections.
-    :rtype: Dict[str, List[airflow.models.connection.Connection]]
+    :rtype: Dict[str, airflow.models.connection.Connection]
```
Comment (mik-laj) [inline airflow/secrets/local_filesystem.py:257]: This is the most important part. This function does not need to return a list. because conn_id is unique.

**Comment by potiuk** [comment]
I believe this is a breaking change - should not we mention it in UPDATING.md ?

**Comment by mik-laj** [comment]
> I believe this is a breaking change - should not we mention it in UPDATING.md ?

This is part of LocalFilesystemBackend and nothing about the behavior of this backend has been changed.  No public API has changed. This  is a private method, although it has not been marked that way and I didn't expect the user to use it. We already have a very long list of entries in UPDATING.md, and I think it should only include those that are important to the user.

However, if you think otherwise, I can add backward compatibility and we will delete these features in a future release.

**Comment by mik-laj** [comment]
@potiuk. I added backward compatibility. We have two similar functions: load_connection and load_connections_dict, but I would like all the functions responsible for loading connections and variables to be moved to a separate module in a separate PR. It will then also rename the method from load_connection_dict to load_connection.

**Review by potiuk — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Pull request merged**